tests/backup/backup_helper.go

```diff
@@ -1490,3 +1493,54 @@ func IsPresent(dataSlice interface{}, data interface{}) bool {
 	return false
 
 }
+
+// return current stork image.
+func getStorkImage() (string, error) {
+	storkDeployment, err := apps.Instance().GetDeployment(storkDeploymentName, storkDeploymentNamespace)
+	if err != nil {
+		return "", err
+	}
+	storkImage := storkDeployment.Spec.Template.Spec.Containers[0].Image
+	return storkImage, nil
+}
+
+// Upgrade the stork to required version.
+func upgradeStorkVersion(reqStorkImage string) (v1.ConditionStatus, error) {
+
+	currentStorkImage, _ := getStorkImage()
+	currentStorkVersionStr := strings.Split(currentStorkImage, ":")[len(strings.Split(currentStorkImage, ":"))-1]
+	currentStorkVersion, _ := version.NewSemver(currentStorkVersionStr)
+
+	reqStorkVersionStr := strings.Split(reqStorkImage, ":")[len(strings.Split(reqStorkImage, ":"))-1]
+	reqStorkVersion, _ := version.NewSemver(reqStorkVersionStr)
+
+	log.Infof("Current stork version : %s", currentStorkVersion)
+	log.Infof("Upgrading stork version to : %s", reqStorkVersion)
+
+	if currentStorkVersion.GreaterThanOrEqual(reqStorkVersion) {
+		log.Debugf("Skipping stork upgrade as current stork version is greater than or equal to [%s]", reqStorkVersion)
+		return v1.ConditionFalse, nil
+	}
+
+	storkDeployment, err := apps.Instance().GetDeployment(storkDeploymentName, storkDeploymentNamespace)
+	if err != nil {
+		return v1.ConditionFalse, err
+	}
+
+	storkDeployment.Spec.Template.Spec.Containers[0].Image = reqStorkImage
```

Comment:
Isn't it *storkDeployment.Spec.Template.Spec.Containers[0].Image = reqStorkImage?